config.mak.uname

```diff
@@ -424,6 +424,8 @@ ifeq ($(uname_S),Windows)
 	NO_POSIX_GOODIES = UnfortunatelyYes
 	NATIVE_CRLF = YesPlease
 	DEFAULT_HELP_FORMAT = html
+	NO_DEFLATE_BOUND = YesPlease
+	# the zlib.c deflateBound, compiled for windows, is limited to uLong 32bit sourceLen.
```

Comment:
Ours DOES NOT have a 64 bit version! It is only sizeof(long) (provided by the same failing zlib library... 
This then cascades into the macro defining that functions worst case value as we can't have the wrong casts in the calls.

Reply:
Oops shouldn't shout. It was just one of those 'learned the hard way' nasties I caught ;-)

Reply:
> Ours DOES NOT have a 64 bit version!

By "ours", I referred to "Git's", i.e. to [this macro](https://github.com/git/git/blob/v2.21.0/zlib.c#L149-L151):

```c
#if defined(NO_DEFLATE_BOUND) || ZLIB_VERNUM < 0x1200
#define deflateBound(c,s)  ((s) + (((s) + 7) >> 3) + (((s) + 63) >> 6) + 11)
#endif
```

As it is a macro, it inherits the data type from whatever is passed into it. So on 64-bit systems, when you pass in a `size_t` argument, it *is* 64-bit.

So I think we pretty successfully did the old white male thing that is so common these days and talked right past each other.

What I meant by saying

> This is a bit misleading. The `Makefile` specifically says that you should set this if your zlib does not have deflateBound. But ours _does_...

was that the comment should read more like this:

```Makefile
# Even if the zlib of Git for Windows' SDK provides a `deflateBound()`,
# we cannot use it because it operates on `unsigned long` parameters,
# which are limited to 32-bit even on 64-bit Windows. Therefore, let's
# pretend that we do not have a `deflateBound()` at all, and force Git
# to fall back to its data-type agnostic macro `deflateBound()` that
# handles 64-bit `size_t` well.
NO_DEFLATE_BOUND = YesPlease
```

Reply:
That's true. aside: I'd got confused between your comment being in the c file where it could be confused with the #define, and the Makefile (config.mak?) where it's a comment..

the other factor was that at one stage there was a cast buried in the call stack which then would pass through the macro to give the wrong answer. One needed to be careful not to screw up https://github.com/git/git/blob/v2.21.0/zlib.c#L153-L156 which has no casts when the macro replacement hits. (it may not be visible now)